docs/source/developer/release.rst

```diff
@@ -0,0 +1,51 @@
+Numba Release Checklist Templates
+==================================
+
+The goal of the Numba release process -- from a birds eye perspective -- is to
+publish source and binary artifacts that correspond to the given version
+number. This usually involves a sequence of individual tasks that must be
+performed in the correct order and with diligence. Numba and llvmlite are
+commonly released in lockstep since there is usually a one-to-one mapping
+between a Numba version and a corresponding llvmlite version.
+
+This section contains various notes and templates that can be used to create a
+Numba release checklist on the Numba Github issue tracker. This is the aid for
+the maintainers during the release process and helps to ensure that all tasks
+are completed in the correct order and that no tasks are accidentally omitted.
+
+Historically, the release checklists were copied from one release to the next.
+While this worked OKish for many years, often times tasks were lost during
+repeated copy and paste. As a result this document was created to keep track of
```

Comment:
Is the history needed?

Reply:
not sure, maybe not, it was mostly documenting my personal motivation, I can remove it if just adds overhead.

Reply:
I'm not sure it's necessary, a good reason for this process is presented in the previous paragraph. I'd rather spend effort keeping documentation such as this accurate with respect to changes/updates, if you feel differently am happy to discuss.

Reply:
sure, no need for that perspective here.